file name encryption/AES-SIV-512-B64URL.md

```diff
@@ -79,6 +79,66 @@ let dirPath = vaultRoot + '/d/' + dirIdString[0..2] + '/' + dirIdString[2..32]
 
 Regardless of the hierarchy of cleartext paths, ciphertext directories are always stored in a flattened structure. All directories will therefore effectively be siblings (or cousins, to be precise).
 
+```mermaid
+---
+title: Overview Mapping Directory IDs to Paths, Encryption of Directory Names and Directory Metadata
+---
+flowchart TD
+   decision{root?}
+   decision -->|n| latestSeed
+   decision -->|n| csprng32
+   decision -->|y| initialSeed
+   latestSeed --> directorySeed
+   csprng32 --> dirId
+   csprng32{{"csprng(32)"}}
+   initialSeed --> directorySeed
+   initialSeed -->|seed:| kdfRootDirId
+   kdfRootDirId{{"kdf(seed,32,'rootDirId')"}}
+   kdfRootDirId --> dirId
+   directorySeed -->|seed:| kdfSiv
+   kdfSiv{{"kdf(seed,64,'siv')"}}
+   kdfSiv --> sivKey
+   directorySeed -->|seed:| kdfHmac
+   kdfHmac{{"kdf(seed,32,'hmac')"}}
+   kdfHmac --> hmacKey
+   hmacKey -->|key:| hmacSha256
+   hmacSha256{{hmacSha256}}
+   hmacSha256 --> dirIdHash
+   dirIdHash --> truncate
+   truncate{{"_[0..20]"}}
+   truncate --> base32
+   base32{{base32}}
+   base32 --> dirIdString
+   dirIdString --> head
+   head{{"_[0..2]"}}
+   dirIdString --> tail
+   tail{{"_[2..32]"}}
+   head -->|$0:| pattern
+   tail -->|$1:| pattern
+   pattern{{"'d/'$0'/'$1"}}
+   pattern --> dirPath
+   sivKey -->|sivKey:| aesSiv
+   aesSiv{{aesSiv}}
+   parentDirId -->|ad:| aesSiv
+   clearTextName -->|secret:| aesSiv
+   aesSiv --> base64Url
+   base64Url{{base64url}}
+   base64Url --> join
+   uvf --> join
+   uvf["'.uvf'"]
+   join{{"join"}}
+   join --> ciphertextName
+   directorySeed --> concat
+   dirId --> concat
+   concat{{concat}}
```

Comment:
This is incorrect. We do not concatenate seed and dirId. Instead we create a normal encrypted file whose sole content is the `dirId`. The seed is used in the file header (as correctly indicated by `directorySeed -->|seed:| fileContentEncryption`)

Reply:
I was mislead by
> Every directory requires certain metadata that affects the file name encryption of its direct children:
> 
> * the seed used to derive keys
> * the directory ID
into thinking that both go in to the body. However, it's actually clear later on:
> The cleartext content of `dir.uvf` is the 32 byte dirId.

Reply:
Done: 5ca480306b88a46cf711b845e89a2d5bdd2dcdb9